app/models/user.rb

```diff
@@ -17,4 +19,31 @@ class User < ApplicationRecord
   def liked?(post_id)
     likes_given.where('post_id = ?', post_id).any?
   end
+
+  def pending_requests
+    Friendship.where('friend_id = ? AND confirmed = ?', id, false)
+  end
```

Comment:
- [ ] User#pending_friends can be defined as association with SQL query instead of processing objects in application memory (as the database was created to store and query data it is always more efficient than ruby) -- The same concept also applies for `sent_requests`